packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts

```diff
@@ -1994,23 +2047,32 @@ const useEchartsCartesianConfig = (
         // Apply border radius to stacked bar charts (only regular stacked, not 100%)
         const isStack100 =
             validCartesianConfig?.layout?.stack === StackType.PERCENT;
+
+        const isStackNone =
+            validCartesianConfig?.layout?.stack === StackType.NONE;
+
         const stackedBarSeries = seriesWithValidStack.filter(
-            (s) => s.type === CartesianSeriesType.BAR && s.stack && !isStack100,
+            (s) =>
+                s.type === CartesianSeriesType.BAR &&
+                s.stack &&
+                !isStack100 &&
+                !isStackNone,
```

Comment:
we only need to calculate the rounded bars on the top-most (or right-most) stack when the bar chart is stacked